panic when running cnitool with Cilium

```core@ip-10-3-0-12 /opt/cni/bin $ sudo -E ./cnitool add cilium /proc/21409/ns/net
DEBU[0000] ADD &{cni /proc/21409/ns/net eth0  /opt/cni/bin {
    "name": "cilium",
    "type": "cilium-cni",
    "mtu": 1450
}
} 
panic: runtime error: slice bounds out of range

goroutine 1 [running, locked to thread]:
github.com/cilium/cilium/common/plugins.SetupVeth(0xc42000c190, 0x3, 0x5aa, 0xc420078ea0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/common/plugins/add_endpoint.go:45 +0xf29
main.cmdAdd(0xc42015bb90, 0x0, 0x0)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/plugins/cilium-cni/cilium-cni.go:297 +0x650
github.com/cilium/cilium/vendor/github.com/containernetworking/cni/pkg/skel.(*dispatcher).checkVersionAndCall(0xc4203dc140, 0xc42015bb90, 0x1693500, 0xc420315050, 0xf9f9f0, 0x0, 0x40)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/vendor/github.com/containernetworking/cni/pkg/skel/skel.go:162 +0x19f
github.com/cilium/cilium/vendor/github.com/containernetworking/cni/pkg/skel.(*dispatcher).pluginMain(0xc4203dc140, 0xf9f9f0, 0xf9f9f8, 0x1693500, 0xc420315050, 0x73d95d1e)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/vendor/github.com/containernetworking/cni/pkg/skel/skel.go:173 +0x2f8
github.com/cilium/cilium/vendor/github.com/containernetworking/cni/pkg/skel.PluginMainWithError(0xf9f9f0, 0xf9f9f8, 0x1693500, 0xc420315050, 0xc420058058)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/vendor/github.com/containernetworking/cni/pkg/skel/skel.go:210 +0xed
github.com/cilium/cilium/vendor/github.com/containernetworking/cni/pkg/skel.PluginMain(0xf9f9f0, 0xf9f9f8, 0x1693500, 0xc420315050)
	/tmp/cilium-net-build/src/github.com/cilium/cilium/vendor/github.com/containernetworking/cni/pkg/skel/skel.go:222 +0x56
main.main()
	/tmp/cilium-net-build/src/github.com/cilium/cilium/plugins/cilium-cni/cilium-cni.go:66 +0x51
netplugin failed but error parsing its diagnostic message "": unexpected end of JSON input```

--- a/common/plugins/add_endpoint.go
+++ b/common/plugins/add_endpoint.go
@@ -33,7 +33,20 @@ const (
 
 // Endpoint2IfName returns the host interface name for the given endpointID.
 func Endpoint2IfName(endpointID string) string {
-	return hostInterfacePrefix + endpointID[:5]
+	return hostInterfacePrefix + truncateString(endpointID, 5)
+}
+
+// Endpoint2TempIfName returns the temporary interface name for the given
+// endpointID.
+func Endpoint2TempIfName(endpointID string) string {
+	return temporaryInterfacePrefix + truncateString(endpointID, 5)
+}
+
+func truncateString(epID string, maxLen uint) string {
+	if maxLen <= uint(len(epID)) {
+		return epID[:maxLen]
+	}
+	return epID
 }
 
 // SetupVeth sets up the net interface, the temporary interface and fills up some endpoint
@@ -41,9 +54,12 @@ func Endpoint2IfName(endpointID string) string {
 // veth, a pointer for the temporary link, the name of the temporary link and error if
 // something fails.
 func SetupVeth(id string, mtu int, ep *models.EndpointChangeRequest) (*netlink.Veth, *netlink.Link, string, error) {
+	if id == "" {
+		return nil, nil, "", fmt.Errorf("invalid: empty ID")
+	}
 
 	lxcIfName := Endpoint2IfName(id)
-	tmpIfName := temporaryInterfacePrefix + id[:5]
+	tmpIfName := Endpoint2TempIfName(id)
 
 	veth := &netlink.Veth{
 		LinkAttrs: netlink.LinkAttrs{Name: lxcIfName},

